### ISSUE TITLE
Installing on Arduino Due to Control late model car?

### ISSUE THREAD

**Issue by jjwallace** [collin80/GEVCU#100]
Hello, me again.
We are converting a late model car which has a stock ECU and would like to replace it with the GEVCU.
Can it control ABS?
Can it export data to the cluster and speedometer so that it works?
Are these pins correct for Arduino?  Pins 9/10 are canbus high and low?

> Pin 1 - +12V Input
> Pin 3 - +12V output for first digital input
> Pin 7 - EXT GND connection to chassis ground / Battery - (in my case)
> Pins 9 & 10 - First canbus
> Pin 19 - Gnd connection for first and second analog inputs (Throttle)
> Pin 20 - First analog input (Throttle)
> Pin 21 - Second analog input (Throttle)
> Pin 22 - Third analog input (brake / testing input for me)
> Pin 23 - Fourth analog input (test pot on my wires)
> Pin 24 - +5V power for first and second analog inputs (Throttle)
> Pin 26 - +5V power for third and fourth analog inputs (test pots for me,
> probably brake transducer for you)
> Pin 30 - Gnd connection for third and fourth analog inputs (Brake / Testing)
> Pin 32 - First digital input. Tied to pin 3 above through a switch
> 
> I used a gnd at pin 19 for one of the gnds. It really doesn't matter
> which one you use. All four (19,29,30,31) are equivalent.

**Comment by collin80** [comment]
Yes, 9 is high, 10 is low for the first CAN bus. All of the grounds are connected and equivalent. You can export data to your custer and speedo so long as they are CAN based. This is very likely on a late model car. Also, you could control ABS so long as those signals were all routed over CAN. I can't say for sure whether this is true or not. I know that late model cars route an awful lot over CAN so it is possible. But, I can't vouch for every vehicle. For both ABS and the cluster/speedo your best bet is to first get many captures from a live, running car to see if the data you're looking for is where you think it is. Here is a link to a bunch of community edited DBC files that may have data that you could use: https://github.com/commaai/opendbc
